src/main/java/undo/UndoController.java

```diff
@@ -48,28 +48,40 @@ public void undoCallback() {
      */
     public void addAction(TurboIssue issue, Action<TurboIssue> action) {
         undoBuffer = Optional.of(new Pair<>(issue, action));
-        action.act(issue).thenApply(success -> showErrorDialogOnFailure(success, issue, action));
-        Notification notification = new Notification(createInfoOcticon(),
-                action.getDescription() + " for #" + issue.getId() + ": " + issue.getTitle(),
-                "Undo", this::undoCallback);
-        notificationController.showNotification(notification);
+        action.act(issue).thenApply(success -> handleActionResult(issue, action, success, false));
     }
 
-    private boolean showErrorDialogOnFailure(Boolean success, TurboIssue issue, Action action) {
+    private boolean handleActionResult(TurboIssue issue, Action<TurboIssue> action, Boolean success,
```

Comment:
Do write javadoc comments for this method and `showNotification`, `showErrorDialog`.